kimchi/src/circuits/polynomials/generic.rs

```diff
@@ -207,7 +207,8 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
 
             // alpha
             let alpha_pow = {
-                let mut res = self.l04.clone();
+                let precomputation = self.get_precomputation();
+                let mut res = precomputation.constant_1_d4.clone();
```

Comment:
it'd be nice not to have to call `get_precomputation` every time. I think it would be nicer if we could do this:

```rust
let mut res = self.precomputations().constant_1_d4()
```